content/en/docs/concepts/services-networking/service.md

```diff
@@ -150,90 +189,129 @@ Every node in a Kubernetes cluster runs a `kube-proxy`. `kube-proxy` is
-faster and more reliable than the userspace proxy. However, unlike the
-userspace proxier, the iptables proxier cannot automatically retry another
-`Pod` if the one it initially selects does not respond, so it depends on
-having working [readiness probes](/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/#defining-readiness-probes).
+In this mode, kube-proxy watches the Kubernetes control plane for the addition and
+removal of Service and Endpoint objects. For each Service, it installs
+iptables rules which capture traffic to the Service's `clusterIP` (which is
+virtual) and `port` and redirects that traffic to one of the Service's
+backend sets.  For each Endpoint object, it installs iptables rules which
+select a backend Pod.
+
+By default, kube-proxy in iptables mode chooses a backend at random.
+
+Using iptables to handle traffic has a lower system overhead, because traffic
+is handled by Linux netfilter without the need switch between userspace and the
+kernel space. This approach is also likely to be more reliable.
+
+If kube-proxy is running in iptables mode and the first Pod that's selected
+does not respond, the connection will fail. This is different from userspace
+mode: in that scenario, kube-proxy would detect that the connection to the first
+Pod had failed and would automatically retry with a different backend Pod.
+
+You can use Pod [readiness probes](/docs/concepts/workloads/pods/pod-lifecycle/#container-probes)
+to verify that backend Pods are working OK, so that kube-proxy in iptables mode
+only sees backends that test out as healthy. Doing this means means you avoid
+having traffic sent via kube-proxy to a Pod that's known to have failed.
 
 ![Services overview diagram for iptables proxy](/images/docs/services-iptables-overview.svg)
 
-### Proxy-mode: ipvs
+### `ipvs` proxy mode {#proxy-mode-ipvs}
 
 {{< feature-state for_k8s_version="v1.11" state="stable" >}}
 
 In this mode, kube-proxy watches Kubernetes Services and Endpoints,
-calls `netlink` interface to create ipvs rules accordingly and syncs ipvs rules with Kubernetes
-Services and Endpoints periodically, to make sure ipvs status is
-consistent with the expectation. When Service is accessed, traffic will
-be redirected to one of the backend Pods.
+calls `netlink` interface to create ipvs rules accordingly and synchronizes
+ ipvs rules with Kubernetes Services and Endpoints periodically.
+This control loop ensures that ipvs status matches the desired
+state.
+When accessing a Service, ipvs will direct traffic to one of the backend Pods.
+
+The ipvs proxy mode is based on netfilter hook function that is similar to
+iptables mode, but uses hash table as the underlying data structure and works
+in the kernel space.
+That means kube-proxy in ipvs mode redirects traffic with a lower latency than
+kube-proxy in iptables mode, wih much better performance when synchronising
+proxy rules. ipvs mode also supports a higher throughput of network traffic.
```

Comment:
should we add smicolon -> `proxy rules; ipvs mode also supports a`

Reply:
I'll reword. Aim here is to write in a way that lends itself to localization.